### PR TITLE
Replaced numbers with SystemDefaultsQos()

### LINKED ISSUE #2225: Use rclcpp::SystemDefaultsQoS() instead of magic numbers (or document magic numbers)
In many places throughout the codebase, we're still using magic numbers to configure QoS for publishers and subscribers for example:
[here](https://github.com/ros-planning/moveit2/blob/55df0bccd5e884649780b4ceeee80891e563b57b/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor_middleware_handle.cpp#L77), [here](https://github.com/ros-planning/moveit2/blob/55df0bccd5e884649780b4ceeee80891e563b57b/moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp#L1130) or [here](https://github.com/ros-planning/moveit2/blob/55df0bccd5e884649780b4ceeee80891e563b57b/moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp#L186).

I think we should replace this in all cases with rclcpp::SystemDefaultsQoS() for consistency and maybe more stability. @henningkayser + @tylerjw What do you think?

### PR DESCRIPTION
**Pull request by Shobuj-Paul**
### Description

This PR replaces "magic numbers" with SystemDefaultsQos(), making it more readable and possibly more stable.
Fixes #2225

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by sjahr** [comment]
Thanks for starting this @Shobuj-Paul. Ping me once the PR is ready for review!

**Review by sjahr — COMMENTED** [review]
This looks good so far

**Comment by Shobuj-Paul** [comment]
Hi @sjahr 
Can I leave instances such as [this](https://github.com/ros-planning/moveit2/blob/6eebe6c47d8a1951efe6b5973d3b8017e47b652a/moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp#L126) untouched? This seems to be more explanatory than just a number.

**Comment by sjahr** [comment]
> Hi @sjahr Can I leave instances such as [this](https://github.com/ros-planning/moveit2/blob/6eebe6c47d8a1951efe6b5973d3b8017e47b652a/moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp#L126) untouched? This seems to be more explanatory than just a number.

Good question :thinking: Can you add a comment to explain this choice (if you know it)? Otherwise just add a `TODO(): ?` comment and we'll figure it out in the review of this PR. @tylerjw Do you know this?

**Comment by Shobuj-Paul** [comment]
> > Hi @sjahr Can I leave instances such as [this](https://github.com/ros-planning/moveit2/blob/6eebe6c47d8a1951efe6b5973d3b8017e47b652a/moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp#L126) untouched? This seems to be more explanatory than just a number.
> 
> Good question thinking Can you add a comment to explain this choice (if you know it)? Otherwise just add a `TODO(): ?` comment to mark it and we'll figure it out in the review of this PR. @tylerjw Do you know this?

Understood, I shall comment on all such instances.

**Comment by Shobuj-Paul** [comment]
@sjahr 
This PR is ready for review. Couple of things left to be confirmed during review:
- Documenting a reason for giving [this](https://github.com/ros-planning/moveit2/blob/6eebe6c47d8a1951efe6b5973d3b8017e47b652a/moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp#L126) kind of QoS profile.
- For python nodes, I couldn't find any equivalent of SystemDefaultsQoS(), so if there a better way of increasing readability of instances like [these](https://github.com/ros-planning/moveit2/blob/cdf7a98689b1400c343417a7f7d35089287a0dfd/moveit_py/moveit/servo_client/teleop.py#L63)?

File: moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp [hunk 1]
```diff
@@ -123,7 +123,7 @@ bool SynchronizedStringParameter::waitForMessage(const rclcpp::Duration& timeout
   auto const nd_name = std::string(node_->get_name()).append("_ssp_").append(name_);
   auto const temp_node = std::make_shared<rclcpp::Node>(nd_name);
   string_subscriber_ = temp_node->create_subscription<std_msgs::msg::String>(
-      name_, rclcpp::QoS(1).transient_local().reliable(),
+      name_, rclcpp::QoS(1).transient_local().reliable(),  // TODO: ?
```
Comment (sjahr) [inline moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp:126]: @tylerjw @henningkayser @JafarAbdi should this be changed to `rclcpp::SystemDefaultsQoS()` too or can you help us with a good comment that explains this exception?

**Review by sjahr — COMMENTED** [review]
I think this looks good so far :+1: Thank you! I'll approve it once we've got an answer regarding ` rclcpp::QoS(1).transient_local().reliable()`

**Comment by sjahr** [comment]
I think you can explicitly set it via https://docs.ros2.org/foxy/api/rclpy/api/qos.html but if you just don't set it [like in the tutorial](https://github.com/ros2/tutorials/blob/master/rclpy_tutorials/topics/listener_py.py), it seems [to default to the system default](https://github.com/ros2/rclpy/blob/913afa019b0d60e7255bd84247dbeab4735d96ea/rclpy/src/rclpy/subscription.cpp#L51). Can you check if that works?

**Comment by Shobuj-Paul** [comment]
> I think you can explicitly set it via https://docs.ros2.org/foxy/api/rclpy/api/qos.html but if you just don't set it [like in the tutorial](https://github.com/ros2/tutorials/blob/master/rclpy_tutorials/topics/listener_py.py), it seems [to default to the system default](https://github.com/ros2/rclpy/blob/913afa019b0d60e7255bd84247dbeab4735d96ea/rclpy/src/rclpy/subscription.cpp#L51). Can you check if that works? @peterdavidfagan Do you have any advice here?

Yep, this looks good. I'll leave the python nodes as they are for now, since this kind of depth setting seems to be documented.

File: moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp (hunk 1 above)
Comment (henningkayser) [reply moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp:126]: this should not be changed. transient_local takes care of supporting late subscriptions

Comment (sjahr) [reply moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp:126]: @Shobuj-Paul Can you replace the `// TODO: ?`  with something like `// "transient_local" is required for supporting late subscriptions.`

Comment (Shobuj-Paul) [reply moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp:126]: Sure, I'll make the change.

**Review by tylerjw — APPROVED** [review]
This is a good cleanup. Thank you!

**Review by henningkayser — APPROVED** [review]

**Pull request merged**